jax/_src/flatten_util.py

```diff
@@ -12,17 +12,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from collections.abc import Iterable
 import numpy as np
+from typing import Any, Callable, TypeAlias
 
 from jax._src.lax import lax
 from jax._src import dtypes
-from jax._src.tree_util import tree_flatten, tree_unflatten
-from jax._src.util import safe_zip, unzip2, HashablePartial
+from jax._src.tree_util import tree_flatten, tree_unflatten, PyTreeDef, Leaf
+from jax._src.util import safe_zip as zip, unzip2, HashablePartial
+from jax._src.typing import Array
```

Comment:
This new import will require updating the `flatten_util` build rule in `jax/_src/BUILD` to depend on `:typing`.